mlflow/johnsnowlabs/__init__.py

```diff
@@ -807,22 +875,88 @@ def _fetch_deps_from_path(local_model_path):
 
 
 def _unpack_and_save_model(spark_model, dst):
+    print(f'Saving model to {dst}')
+    # dst = dst.replace(':','')
     from pyspark.ml import PipelineModel
+    try:
+        if isinstance(spark_model, _PyFuncModelWrapper):
+            spark_model = spark_model.spark_model
 
-    if isinstance(spark_model, _PyFuncModelWrapper):
-        spark_model = spark_model.spark_model
-    if isinstance(spark_model, PipelineModel):
-        spark_model.write().overwrite().save(dst)
-    else:
-        # nlu pipe
-        spark_model.predict("Init")
+        if isinstance(spark_model, PipelineModel):
+            spark_model.write().overwrite().save(dst)
+            return
+        else:
+            try:
+                # Fit Pipe if not fitted. May fail on complex pipelines with multiple input cols
+                spark_model.predict("Init")
+            except:
+                pass
+            try:
+                spark_model.vanilla_transformer_pipe.write().overwrite().save(dst)
+            except Exception as err:
+                print(err)
+                # for mlflowdbfs_path we cannot use overwrite, gives
+                # org.apache.hadoop.fs.UnsupportedFileSystemException: No FileSystem for scheme
+                # "mlflowdbfs"
+                spark_model.save(dst)
+
+    except Exception as err:
+        print(f"Could  not save your model, use a different path for saving or remove {dst}. \n"
+              f"If this issue persists, please restart your notebook kernel.")
+        raise err
+
+
+def _maybe_unpack_and_save_model(spark_model, dst):
+    try:
+        _unpack_and_save_model(spark_model, dst)
+        return True
+    except:
+        return False
+
+
+import pandas as pd
+from pydantic import BaseModel
```

Comment:
`pydantic` is not an mlflow's dependency. can we use dataclass or namedtuple?

Reply:
@harupy  pydantic is a johnsnowlabs dependency. Any user of this flavor must have it installed anyways. 
If they manage to import mlflow.johnsnowlabs, they must have johnsnowlabs installed.
If johnsnowlabs is installed, there must be pydantic in the env, so I think its ok. 
What do you think?